### ISSUE TITLE
Updated packages for Linux

### ISSUE THREAD

**Issue by stefanct** [endless-sky/endless-sky#4630]
0.9.10 is now out for a while but neither Debian nor the Ubuntu PPA have received an updated version. Also, the release announcement/tag message refers to Debian's mentor packages website which is probably outdated and should rather link to https://packages.debian.org/search?keywords=endless-sky&searchon=names&exact=1&suite=all&section=all or so?

**Comment by JRottm** [comment]
Seconded, please, please update the Debian package.
(It might be better to remove the "documentation" label from this issue, as that really isn't the central point here and might be the reason why the issue gets overlooked?)
Thanks!

**Comment by Pointedstick** [comment]
You need to contact the Debian packagers to get updated packages, which is unlikely for Debian Stable since Debian deliberately ships old packages in the name of "stability." It may get updated in Debian Testing though.

**Comment by stefanct** [comment]
Do you really have nothing better to do than embarrass yourself publicly by flaming on a public issue tracker without having any clue? Michael aka @endless-sky *is* the packager for endless-sky in Debian.

**Comment by MCOfficer** [comment]
> 
> 
> Do you really have nothing better to do than embarrass yourself publicly by flaming on a public issue tracker without having any clue? Michael aka @endless-sky _is_ the packager for endless-sky in Debian.

Let's dial it down a bit please. While, as far as i can see, [you are correct](https://packages.debian.org/buster/endless-sky), an issue tracker is a place of productive discussion - not the kind of flaming you seem to be averse to.

Wether Pointed's answer was perfectly accurate or not, afaik "It's hard and painstaking to get a package into the debian repos" still holds true.

**Comment by Pointedstick** [comment]
Ah, I didn't actually know that @endless-sky was the Debian packager. Thanks for the correction.

Still, the request would ideally be made on Debian infrastructure, not here.

**Comment by endless-sky** [comment]
My Ubuntu PPA is out of storage space, so I need to wait for it to be freed up before I can upload the latest version.

Note that I'm not the Debian uploader; I build the package, but an official Debian developer has to choose to add it to their repository.

**Comment by olekw** [comment]
> My Ubuntu PPA is out of storage space, so I need to wait for it to be freed up before I can upload the latest version.
> 
> Note that I'm not the Debian uploader; I build the package, but an official Debian developer has to choose to add it to their repository.

Debian Developer here and I'm happy to sponsor your upload. I actually sent you an email a few days ago and didn't realize that you're upstream as well. Happy to help however I can! :)

**Comment by temtemy** [comment]
Hi, Debian user here. Is there any update on this? Bullseye might become stable in the future and the endless-sky package is still in 0.9.8... Btw, if you guys are going to package 0.9.12 (or 0.9.13 if that somehow got released before Debian gets up-to-date with upstream) in sid and testing, you might as well package it for stable-backports for us stable users who had been waiting for a 0.9.12 in Debian. :)

**Comment by MCOfficer** [comment]
as of now it's basically stalled. Afaik MZ is the only community member capable of doing this, the rest of us either have no experience with debian or no interest in becoming official maintainer.

FWIW Until MZ comes back, we're more likely to have something like a https://packagecloud.io/ repository than an uptodate debian package.

**Comment by olekw** [comment]
@MCOfficer, I wouldn't say that. I'm happy to coach/mentor anyone who has the desire to update the Debian package. Unfortunately, I'm a little too busy right now to commit to doing it myself but I definitely want an updated package in Debian. It's honestly not that difficult for someone with motivation and decent Linux familiarity. The biggest hurdle for a new contributor is getting a DD to sponsor your upload (seriously, ask anyone, and I've been there myself). Oh, you've already got that base covered. ;)

**Comment by MCOfficer** [comment]
The biggest hurdle in my book is working under my real name, which seems to be a given in "official debian business". If that's not necessary, i could possibly seize the mantle.

**Comment by olekw** [comment]
@MCOfficer, I'm not aware of any such requirement, not even for Debian Developers. Debian takes privacy seriously and there are some people who work in more dangerous parts of the world where they may not wish to disclose their identity publicly. With that being said, there is an expectation of having a verifiable online identity and of *someone* in Debian knowing who you actually are, even if that information is not disclosed to anyone else. After all, we want to know "who" is making changes. Anonymous contributions are not a thing.

So, tl;dr: real name not required. Send me an email and let's discuss specifics. First name and the org I'm a part of.

**Comment by srkunze** [comment]
Does somebody know a working ppa of endless-sky? :-)

**Comment by neevok-eu** [comment]
Any chance for a new linux update as well as a manual way to update without any package managers?

Would it be possible to adapt the Linux version for android? Not any input/gui changes, just make it installable and playable with a keyboard/mouse.

**Comment by MCOfficer** [comment]
> Any chance for a new linux update as well as a manual way to update without any package managers?

As a manual way, you can always self-compile or use the provided AppImage.

> Would it be possible to adapt the Linux version for android? Not any input/gui changes, just make it installable and playable with a keyboard/mouse.

As of now: we can't rule out the possibility, but won't make any official effort in that direction, see #1502. The most practical solution is probably the unofficial [Endless Web Project](https://play-endless-web.com/).

**Comment by olekw** [comment]
On Mon, Jan 31, 2022, 04:59 MCOfficer ***@***.***> wrote:

> Any chance for a new linux update as well as a manual way to update
> without any package managers?
>
> As a manual way, you can always self-compile or use the provided AppImage.
>

Renewing my offer on the Debian package front. If you or anyone else will
agree to maintain the package, I'd be happy to help with the initial
creation and sponsoring.

That would take care of every Debian-based Linux distro (including all
Ubuntu-derivatives)

**Comment by neevok-eu** [comment]
> > Any chance for a new linux update as well as a manual way to update without any package managers?
> 
> As a manual way, you can always self-compile or use the provided AppImage.
> 
> > Would it be possible to adapt the Linux version for android? Not any input/gui changes, just make it installable and playable with a keyboard/mouse.
> 
> As of now: we can't rule out the possibility, but won't make any official effort in that direction, see #1502. The most practical solution is probably the unofficial [Endless Web Project](https://play-endless-web.com/).

Thanks for the tip about the web based version. I will try it on my Planet Computers Astro Slide as soon as I get it.

It is also helpful to know that the current version can be compiled for Linux and the AppImage can also be used to play the latest version on Linux. I will try both, though it will be the first time doing something like that in Linux for me. Is there a specific instruction for this or can I just use any generic instruction for compiling in Linux and using an AppImage?
TIA

**Comment by MCOfficer** [comment]
> It is also helpful to know that the current version can be compiled for Linux and the AppImage can also be used to play the latest version on Linux. I will try both, though it will be the first time doing something like that in Linux for me. Is there a specific instruction for this or can I just use any generic instruction for compiling in Linux and using an AppImage? TIA

the AppImage is just an executable that only requires some dependencies which are usually preinstalled (FUSE, for example). There's nothing special about ES's AppImage, so any [generic instructions](https://docs.appimage.org/user-guide/run-appimages.html) will do. As for compiling, see [this wiki page](https://github.com/endless-sky/endless-sky/wiki/BuildInstructions#linux).

**Comment by neevok-eu** [comment]
Thanks, I will try both and try to make a step by step guide for others so they can reproduce it. I was looking for something like that in this forum.

**Comment by ziproot** [comment]
https://packages.debian.org/sid/endless-sky
Looks like it's been updated at least for unstable.

**Comment by TomGoodIdea** [comment]
We don't maintain officially any deb packages anymore, so this issue should be reported to the administrators of particular repositories. As far as I know, Debian maintainers are aware of this problem, and are not updating their package because of some licensing issues (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1061241).
On our side, there are #9355 and #9413, so any further discussion should be moved there.
